source/common/common/fancy_logger.cc

```diff
@@ -0,0 +1,118 @@
+#include <atomic>
+#include <memory>
+
+#include "common/common/logger.h"
+
+using spdlog::level::level_enum;
+
+namespace Envoy {
+
+/**
+ * FancyLogger implementation: a logger with fine-grained log control.
+ */
+
+/**
+ * If the sink is initialized.
+ */
+int kSinkInit = 0;
+
+/**
+ * Lock for the following global map (not for the corresponding loggers).
+ */
+absl::Mutex fancy_log_lock__;
```

Comment:
This is a funny name. Find a proper naming convention here? https://google.github.io/styleguide/cppguide.html#Naming

Reply:
Currently it's fancy_log_lock. Should be fine?